### PR TITLE
adding md5 and gitignore support

### PR DESCRIPTION
**Pull request by dgraeber**
*Issue #, if available:*
#58
*Description of changes:*

- Add support for gitignore when calculating the module md5 of the source code
- Add support to exclude standard module files from md5 if they don't impact code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: seedfarmer/checksum.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            ignore_maps[ignore_path] = parse_gitignore(ignore_path)
+
+    return ignore_maps
+
+
+def _evaluate_file(filename: str, ignore_maps: Dict[str, Any]) -> bool:
+    if ignore_maps is None:
+        return False
+    for ignore_key in ignore_maps.keys():
+        if ignore_maps.get(ignore_key):
```
[the remark this answers is not in the document]
Comment (dgraeber) [reply seedfarmer/checksum.py:46]: I am making sure that the value returned is not None.  When I initially populated the map, I did not have the check to see if the .`gitignore` path existed...so the value was None for the key.  This led errors later on...

I can remove this check now

File: seedfarmer/checksum.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            inIgnore = (ignore_maps.get(ignore_key))(filename)  # type: ignore
```
[the remark this answers is not in the document]
Comment (dgraeber) [reply seedfarmer/checksum.py:47]: I must have blind habits...will correct

File: seedfarmer/checksum.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    _get_paths(os.path.join(project_path, module_path))
+    for ignore_path in ignore_paths:
+        if os.path.exists(ignore_path):
+            ignore_maps[ignore_path] = parse_gitignore(ignore_path)
+
+    return ignore_maps
+
+
+def _evaluate_file(filename: str, ignore_maps: Dict[str, Any]) -> bool:
```
[the remark this answers is not in the document]
Comment (dgraeber) [reply seedfarmer/checksum.py:42]: Yeah...I had some docs explaining the code but removed them b/c my comments seem to flag issues from other reviewers.  I agree and can add them back...

The values are callables (they represent lambdas)

File: seedfarmer/checksum.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    def _get_paths(working_dir: str) -> None:
+        gitignore_path = os.path.join(working_dir, ".gitignore")
+        ignore_paths.append(gitignore_path) if os.path.exists(gitignore_path) else None
+        if os.path.realpath(working_dir) != os.path.realpath(project_path):
+            _get_paths(str(Path(os.path.join(working_dir, os.pardir)).resolve()))
+
+    _get_paths(os.path.join(project_path, module_path))
+    for ignore_path in ignore_paths:
+        if os.path.exists(ignore_path):
+            ignore_maps[ignore_path] = parse_gitignore(ignore_path)
```
[the remark this answers is not in the document]
Comment (dgraeber) [reply seedfarmer/checksum.py:37]: it is a lambda...I can add comments

File: seedfarmer/checksum.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chamcca) [reply seedfarmer/checksum.py]: if the check is valid and the key can have a value of None, then leave it. though i would be explicit with an `is not None`

**Review by chamcca — APPROVED** [review]

**Pull request merged**